### ISSUE TITLE
Support setting EC2 instance metadata to require token (IMDSv2)

### ISSUE THREAD

**Issue by alex** [terraform-providers/terraform-provider-aws#10949]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
Launched here: https://aws.amazon.com/blogs/security/defense-in-depth-open-firewalls-reverse-proxies-ssrf-vulnerabilities-ec2-instance-metadata-service/

EC2 instances can be set to require IMDSv2 only. This can be done to running instances with https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#EC2.ModifyInstanceMetadataOptions or new instances with the `MetadataOptions` field of `RunInstancesInput`.

### New or Affected Resource(s)

- `aws_instance`

### Potential Terraform Configuration

```hcl
resource "aws_instance" "this" {
    // ...
    metadata_options {
        http_tokens = "required"
    }
}
```

**Comment by atpoirie** [comment]
This should also support optionally setting the PUT response hop limit

Such as outlined by the following AWS CLI command
aws ec2 modify-instance-metadata-options --instance-id i-1234567898abcdef0 --http-put-response-hop-limit 3

`
resource "aws_instance" "this" {
    // ...
    metadata_options {
        http_tokens = "required"
        http_put_response_hop_limit = "1"
    }
}
`

**Comment by bflad** [comment]
Support for configuring metadata options in the `aws_instance` and `aws_launch_template` resources has been merged and will release with version 2.55.0 of the Terraform AWS Provider, later today. Thanks to @stefansundin and @ewbankkit for the implementation. 👍
